scripts/generate_icon_types.js

```diff
@@ -0,0 +1,234 @@
+ * @param {string[]} solidIcons
+ * @returns {string[]} - Sorted unique icon names
+ */
+function getUniqueIconNames(outlineIcons, solidIcons) {
+  const uniqueSet = new Set([...outlineIcons, ...solidIcons]);
+  return Array.from(uniqueSet).sort();
+}
+
+/**
+ * Generate TypeScript union type from array of strings
+ * @param {string[]} items
+ * @returns {string}
+ */
+function generateUnionType(items) {
+  if (items.length === 0) return 'never';
+  return items.map(item => `"${item}"`).join(' | ');
+}
+
+/**
+ * Generate JavaScript file with icon arrays. AllIcons is computed dynamically from OutlineIcons and SolidIcons.
+ * @param {string[]} outlineIcons
+ * @param {string[]} solidIcons
+ */
+function generateJavaScriptFile(outlineIcons, solidIcons) {
+  const content = `// Auto-generated file. Do not edit manually.
+// Generated from assets/icons directory structure
+
+/**
+ * Array of all outline icon names (from assets/icons/outline)
+ * @type {ReadonlyArray<string>}
+ */
+export const OutlineIcons = ${JSON.stringify(outlineIcons, null, 2)};
+
+/**
+ * Array of all solid icon names (from assets/icons/solid)
+ * @type {ReadonlyArray<string>}
+ */
+export const SolidIcons = ${JSON.stringify(solidIcons, null, 2)};
+
+/**
+ * Array of all unique icon names (union of outline and solid)
+ * Computed dynamically to avoid duplication and reduce package size
+ * @type {ReadonlyArray<string>}
+ */
+export const AllIcons = Array.from(new Set([...OutlineIcons, ...SolidIcons])).sort();
+
+const outlineIconsSet = new Set(OutlineIcons);
+/**
+ * Check if an icon name exists in the outline set
+ * @param {string} iconName - Icon name to check
+ * @returns {boolean}
+ */
+export function isOutlineIcon(iconName) {
+  return outlineIconsSet.has(iconName);
+}
+
+const solidIconsSet = new Set(SolidIcons);
+
+/**
+ * Check if an icon name exists in the solid set
+ * @param {string} iconName - Icon name to check
+ * @returns {boolean}
+ */
+export function isSolidIcon(iconName) {
+  return solidIconsSet.has(iconName);
+}
+
+const allIconsSet = new Set(AllIcons);
+
+/**
+ * Check if an icon name exists in any set
+ * @param {string} iconName - Icon name to check
+ * @returns {boolean}
+ */
+export function isValidIcon(iconName) {
+  return allIconsSet.has(iconName);
+}
+`;
+
+  writeFileSync(outputJsPath, content, 'utf8');
+}
+
+/**
+ * Generate TypeScript definition file
+ * @param {string[]} outlineIcons
+ * @param {string[]} solidIcons
+ * @param {string[]} allIcons
+ */
+function generateTypeScriptFile(outlineIcons, solidIcons, allIcons) {
+  const outlineType = generateUnionType(outlineIcons);
+  const solidType = generateUnionType(solidIcons);
+  const allIconsType = generateUnionType(allIcons);
+
+  const content = `// Auto-generated file. Do not edit manually.
+// Generated from assets/icons directory structure
+
+/**
+ * Array of all outline icon names (from assets/icons/outline)
+ */
+export declare const OutlineIcons: readonly string[];
```

Comment:
considere utilizar nesses casos os tipos literais, para indicar quais strings estão nesse array (já que você criou os tipos)

```suggestion
export declare const OutlineIcons: readonly OutlineIcon[];
```